target/linux/generic/files/drivers/platform/mikrotik/mikrotik_wlan_lz77.c

```diff
@@ -0,0 +1,443 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+/*
+ * Copyright (C) 2023 John Thomson
+ */
+
+#include <linux/module.h>
+#include <linux/slab.h>
+#include <linux/string.h>
+#include <linux/errno.h>
+
+#include "mikrotik_wlan.h"
+
+#define MIKRO_LZ77 "[mikrotik wlan lz77] "
+
+#define MIKRO_LZ77_MAX_ENCODED 0x1000
+#define MIKRO_LZ77_MAX_DECODED 0x40000
+
+/* examples have all decompressed to start with DRE\x00 */
+#define MIKRO_LZ77_EXPECTED_OUT	(('E' << 16) | ('R' << 8) | ('D'))
```

Comment:
Yes, but I don't want the rest that comes with that include.

Reply:
Why? It's only a handful of definitions

Reply:
I moved the test comparison to happen once the first four bytes are output, so that it can fail early and avoid many cycles. If you still don't like the idea, I can cut it.